packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -528,6 +528,7 @@ class StrutStyle with Diagnosticable {
         height != other.height ||
         leading != other.leading ||
         forceStrutHeight != other.forceStrutHeight ||
+        leadingDistribution != other.leadingDistribution ||
```

Comment:
nit: leadingDistribution is only relevant if `height` is not null, consider ignore the equality if `height` is null here and everywhere else.